File: src/main/java/com/ulbs/careerstartup/dto/CompanyDTO.java
package com.ulbs.careerstartup.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;
import org.hibernate.validator.constraints.URL;

import jakarta.validation.constraints.NotNull;
import java.util.Collection;
import java.util.UUID;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
public class CompanyDTO {

    private UUID id;

    @NotNull(message = "Name is required")
    @NotEmpty(message = "Name is required")
    private String name;

    @NotNull(message = "Description is required")
    @NotEmpty(message = "Description is required")
    private String address;

    @NotNull(message = "Website is required")
    @NotEmpty(message = "Website is required")
    @URL(message = "Invalid website")
    private String website;

    private String logoUrl;

    private String description;

    @JsonProperty("jobHistories")
    private Collection<JobHistoryDTO> jobHistoriesDTO;

    @JsonProperty("postedJobs")
    private Collection<PostedJobDTO> postedJobsDTO;

    @JsonProperty("reviews")
    private Collection<ReviewDTO> reviewsDTO;
}

File: src/app/services/models/company-dto.ts
/* tslint:disable */
/* eslint-disable */
import { JobHistoryDto } from '../models/job-history-dto';
import { PostedJobDto } from '../models/posted-job-dto';
import { ReviewDto } from '../models/review-dto';
export interface CompanyDto {
  address: string;
  description?: string;
  id?: string;
  jobHistories?: Array<JobHistoryDto>;
  logoUrl?: string;
  name: string;
  postedJobs?: Array<PostedJobDto>;
  reviews?: Array<ReviewDto>;
  website: string;
}
